src/bucklescript/output_bucklescript_decoder.re

```diff
@@ -152,17 +198,34 @@ let generate_error = (loc, message) => {
 let rec generate_decoder = config =>
   fun
   | Res_nullable(loc, inner) =>
-    generate_nullable_decoder(config, conv_loc(loc), inner)
+    lean_parse()
```

Comment:
what do you think about using `when` guard for `lean_parse` check?

Reply:
Might be better. But if we make this the default soon, and with records all the code will be rewritten anyway, I don't think it makes a big difference.